content/en/platform/corda/5.0-beta/operating/notary-uniqueness-metrics.md

```diff
@@ -0,0 +1,75 @@
+{{< /note >}}
+
+## Ledger Uniqueness Checker Client Service
+
+The context exposed to the ledger uniqueness checker client service is minimal. As a result, only a single metric is captured:
+
+|Metric name|Type|Tags| Description|
+|-------------------------------------|-------|--------------|--------------------------------------------|
+| `ledger.uniqueness.client.run.time` | Timer |`result.type` |The time taken from requesting a uniqueness check to a response being received. The `result.type` tag is set to the relevant simple class name of the specific `UniquenessCheckResult` subclass.|
+
+## Uniqueness Checker
+
+The uniqueness checker handles the business logic of uniqueness checking. The implemenation is batched at two levels and three categories of metrics are provided:
+
+* Metrics starting with `uniqueness.checker.batch` relate to "top level" metrics, which apply to a single batch processed by the uniqueness checker. As a batch may contain requests from different notary services and/or virtual nodes, these metrics provide no context as to the identities of the batch being processed.
+* Metrics starting with `uniqueness.checker.subbatch` relate to "sub-batch level" metrics. Each sub-batch represents a partition for each notary virtual node identity within a batch, and therefore are associated with this context via the existing `virtualnode.source` tag, which is already used in the context of P2P metrics.
+* Metrics starting with `uniqueness.checker.request` relate to metrics applicable to specific requests within a sub-batch, such as the result of a request. These are also tagged with the `virtualnode.source`.
+
+|Metric Name|Type|Tags|Description|
+|-------------------------------|------------------|------------------|------------------------|
+| `uniqueness.checker.batch.execution.time`    | Timer    | None    | The overall execution time for a batch, inclusive of all sub-batches. |
+| `uniqueness.checker.batch.size`    | DistributionSummary    | None    | The number of requests in a batch.  |
+| `uniqueness.checker.subbatch.execution.time`    | Timer    | `virtualnode.source`    | The execution time for a specific sub-batch.  |
+| `uniqueness.checker.subbatch.size`    | DistributionSummary    | `virtualnode.source`    |The number of requests in a sub-batch.  |
+| `uniqueness.checker.request.count`    | Counter    | `virtualnode.source`, `result.type`, `duplicate`   | A count of the number of requests processed. It is useless on its own as this information is already captured at the batch and sub-batch levels, but the tags can be used to provide additional context. The `result.type` tag can be used to understand the number of successful vs failed requests, and the type of failures. The `duplicate` tag is set to `true` if the uniqueness checker has seen a request for this transaction before, and is therefore simply returning the original result. Otherwise, it is `false`.  |
```

Comment:
```suggestion
| `uniqueness.checker.request.count`    | Counter    | `virtualnode.source`, `result.type`, `duplicate`   | A count of the number of requests processed. On its own this simply duplicates information that is already captured at the batch and sub-batch levels, but the tags can be used to provide additional context not available in the other metrics. The `result.type` tag can be used to understand the number of successful vs failed requests, and the type of failures. The `duplicate` tag is set to `true` if the uniqueness checker has seen a request for this transaction before, and is therefore simply returning the original result. Otherwise, it is `false`.  |
```

I think we should avoid strong words such as "useless" to describe our metrics!